### DISCUSSION TITLE
Large gap between pieces

### DISCUSSION THREAD

**Discussion by QuestionDevelopment** [flbulgarelli/headbreaker#79] [Q&A] [answered]
Hello,
My puzzle has large gaps between pieces that I can't get rid of.  I am trying to make a puzzle with 0 borders and 0 gaps between pictures so when assembled it looks like a normal photo.

Attached is a photo of two connected pieces

![Screenshot 2024-04-01 153954](https://github.com/flbulgarelli/headbreaker/assets/394869/1a904e95-5e6d-42aa-b1cc-08909ac64c0a)

Here is the code I am using to generate the puzzle. Image is 480px (width) by 660px (height) if that makes a difference.

`  let jigsaw = new Image();
        jigsaw.src = '/module/zero/image/fingerprint/jigsaw4.png?v=2';
        jigsaw.onload = () => {
            const sound = new headbreaker.Canvas('jigsaw-canvas', {
                width: 640, height: 825,
                pieceSize: {x: 160, y: 165}, proximity: 30,
                borderFill: 0.0, strokeWidth: 0,
                lineSoftness: 0.0, image: jigsaw,
                strokeColor: 'black', preventOffstageDrag: true,
                fixed: true, maxPiecesCount: {x: 3, y: 4}
            });
            //sound.adjustImagesToPuzzleHeight();
            sound.adjustImagesToPuzzleWidth();
            sound.autogenerate({
                horizontalPiecesCount: 3,
                verticalPiecesCount: 4
            });
            sound.shuffle(0.8);
            sound.draw();

            sound.attachSolvedValidator();
            sound.onValid(() => {
                alert("Solved");
            });
            sound.onConnect((_piece, figure, _target, targetFigure) => {
                console.log('connect');
                // play sound
                //audio.play();
                //sound.redraw();
                //setTimeout(() => {
                //sound.redraw();
                //}, 200);
            });

            sound.onDisconnect((it) => {
                console.log('disconnect');
                //audio.play();
            });
`

Lastly, is there any documentation that explains what the parameters do.  For example, I had a complete misunderstanding of what proximity did until I read the thread in your discussion.  It would be nice to things outlined.

Thanks for the great project and any support you can provide.

**Comment by flbulgarelli** [comment, marked as answer]
Hi! Have you tried this sample: https://flbulgarelli.github.io/headbreaker/#perfect-match?

Just use a border fill that is the half of the proximity.

Hope that helps!

PS: please check this answer, too: https://github.com/flbulgarelli/headbreaker/discussions/70
